diff --git a/src/org/tn5250j/My5250.java b/src/org/tn5250j/My5250.java
index 3d5eb61..6928a4c 100644
--- a/src/org/tn5250j/My5250.java
+++ b/src/org/tn5250j/My5250.java
@@ -76,15 +76,14 @@ public class My5250 implements BootListener, SessionListener, EmulatorActionList
 
 	private TN5250jLogger log = TN5250jLogFactory.getLogger(this.getClass());
 
-	My5250 () {
+	public My5250 () {
 
-		splash = new TN5250jSplashScreen("tn5250jSplash.jpg");
-		splash.setSteps(5);
-		splash.setVisible(true);
+        splash = new TN5250jSplashScreen("tn5250jSplash.jpg");
+        splash.setSteps(5);
+        splash.setVisible(true);
 
 		loadLookAndFeel();
 
-
 		loadSessions();
 		splash.updateProgress(++step);
 
@@ -106,6 +105,9 @@ public class My5250 implements BootListener, SessionListener, EmulatorActionList
 		Tn5250jController.getCurrent();
 	}
 
+    public SessionManager getSessionManager() {
+        return manager;
+    }
 
 	/**
 	 * we only want to try and load the Nimbus look and feel if it is not
@@ -451,6 +453,30 @@ public class My5250 implements BootListener, SessionListener, EmulatorActionList
 		return null;
 	}
 
+    public void startNewUbluSession(String [] args) {
+
+        String sel = "";
+
+        if (containsNotOnlyNullValues(sessionArgs) && !sessionArgs[0].startsWith("-")) {
+            sel = sessionArgs[0];
+        } else {
+            sel = getDefaultSession();
+        }
+
+        Sessions sess = manager.getSessions();
+
+        if (sel != null && sess.getCount() == 0 && sessions.containsKey(sel)) {
+            sessionArgs = new String[TN5250jConstants.NUM_PARMS];
+            parseArgs(sessions.getProperty(sel), sessionArgs);
+        }
+
+//		if (sessionArgs == null || sess.getCount() > 0 || sessions.containsKey("emul.showConnectDialog")) {
+//			openConnectSessionDialogAndStartSelectedSession();
+//		} else {
+        newSession("ublu", args);
+//		}
+    }
+
 	private void startNewSession() {
 
 		String sel = "";
@@ -475,7 +501,6 @@ public class My5250 implements BootListener, SessionListener, EmulatorActionList
 		}
 	}
 
-
 	private void openConnectSessionDialogAndStartSelectedSession() {
 		String sel = openConnectSessionDialog();
 		Sessions sess = manager.getSessions();
